python/cudf/cudf/_libxx/datetime.pyx

```diff
@@ -0,0 +1,42 @@
+from libcpp.memory cimport unique_ptr
+
+from cudf._libxx.cpp.column.column cimport column
+from cudf._libxx.cpp.column.column_view cimport column_view
+
+from cudf._libxx.move cimport move
+from cudf._libxx.column cimport Column
+
+cimport cudf._libxx.cpp.datetime as libcudf_datetime
+
+
+def extract_datetime_component(Column col, object field):
+
+    cdef unique_ptr[column] c_result
+    cdef column_view col_view = col.view()
+
+    with nogil:
+        if field == "year":
+            c_result = move(libcudf_datetime.extract_year(col_view))
+        elif field == "month":
+            c_result = move(libcudf_datetime.extract_month(col_view))
+        elif field == "day":
+            c_result = move(libcudf_datetime.extract_day(col_view))
+        elif field == "weekday":
+            c_result = move(libcudf_datetime.extract_weekday(col_view))
+        elif field == "hour":
+            c_result = move(libcudf_datetime.extract_hour(col_view))
+        elif field == "minute":
+            c_result = move(libcudf_datetime.extract_minute(col_view))
+        elif field == "second":
+            c_result = move(libcudf_datetime.extract_second(col_view))
+        else:
+            raise ValueError(f"Invalid datetime field: '{field}'")
+
+    result = Column.from_unique_ptr(move(c_result))
+
+    if field == "weekday":
+        # Pandas counts Monday-Sunday as 0-6
+        # while we count Monday-Sunday as 1-7
+        result = result.binary_operator("sub", result.dtype.type(1))
```

Comment:
Does Spark / anyone else need ISO encoding as opposed to C encoding upstream? Having to do an extra binaryop here is a bit of a shame. cc @trxcllnt

Reply:
The C encoding is fine with me, but @jdye64 might have thoughts re: cudf-java bindings

Reply:
@jlowe you mean ;)

Reply:
Lets raise an issue and discuss

Reply:
Spark uses 1=Sunday 2=Monday, ..., 7=Saturday.  See https://spark.apache.org/docs/latest/api/sql/index.html#dayofweek

Reply:
> Spark uses 1=Sunday 2=Monday, ..., 7=Saturday. See https://spark.apache.org/docs/latest/api/sql/index.html#dayofweek

So we're both wrong according to ISO 😆

Reply:
We can both be wrong in different ways together!